backend/server/importlib/headerSource.py

```diff
@@ -98,8 +98,21 @@ def populate(self, header, size, checksum_type, checksum, path=None, org_id=None
             self['payload_format'] = 'cpio'
         if self['payload_size'] is None:
             self['payload_size'] = 0
+        # Extra tags
+        self._populateExtraTags(header)
+
         return self
 
+    def _populateExtraTags(self, header):
+        """
+        Populate extra tags. Currently only "modularitylabel".
+        """
+        mlabel = header.modularity_label()
+        if mlabel is not None:
+            self["extra_tags"] = [
```

Comment:
I think this is only initialized in case of a `rpmBinaryPackage`. When this is a `rpmSourcePackage` extra_tags is not part of the tagMap. 

I would suggest to do a similar thing as `build_time` which is available in both tagMaps.
Just add `extra_tags` to `rpmSourcePackage` tagMap.

```python
class rpmSourcePackage(SourcePackage, rpmPackage):
    tagMap = rpmPackage.tagMap.copy()
    tagMap.update({
        'package_group': 'group',
        'rpm_version': 'rpmversion',
        'payload_size': 'archivesize',
        'build_host': 'buildhost',
        'build_time': 'buildtime',
        'source_rpm': 'sourcerpm',
        # Arrays: require a different mapping
        # We set them differently
        'package_size': None,
        'org_id': None,
        'md5sum': None,
        'path': None,
        # Unsupported
        'payload_format': None,
        'channels': None,
        'package_id': None,
    })
```

Reply:
@mcalmer Well, to be honest I am not sure why this is at all done that way. Normally, populator should either add stuff or not at all. And so the importer should `.get("extra_tags, [])` instead of checking "if it is None? and if it is non-empty?" and trying to iterate over e.g. `int` or `float` mistakenly put there and thus crash. :wink: Anyway, this is something to cleanup in a future.

For now I've added `"extra_tags": None`, but I think this is a wrong approach.